src/test/java/spoon/test/lambda/LambdaTest.java

```diff
@@ -450,4 +454,17 @@ private void assertIsWellPrinted(String expected, CtLambda<?> lambda) {
 	private CtLambda<?> getLambdaInFooByNumber(int number) {
 		return foo.getElements(new TypeFilter<CtLambda<?>>(CtLambda.class)).get(number);
 	}
+
+	@Test
+	public void testGetDeclarationOnTypeParameterFromLambda() {
+		List<CtTypeParameterReference> listCtTPR  = launcher.getModel().getElements(new TypeFilter<>(CtTypeParameterReference.class));
+
+		for (CtTypeParameterReference typeParameterReference : listCtTPR) {
+			if (!(typeParameterReference instanceof CtWildcardReference) && typeParameterReference.getDeclaration() == null) {
+				System.err.println(typeParameterReference.getSimpleName()+" from parent "+typeParameterReference.getParent(CtClass.class).getPosition()+"  has null declaration");
+				typeParameterReference.getDeclaration();
+				fail();
+			}
```

Comment:
Ok let's finish it, I need it to close another one, we can always improve the tests later :) Thanks for the comments!